session/session.go

```diff
@@ -64,16 +82,77 @@ func (s *Session) Has(key string) bool {
 	return val != nil
 }
 
+func (s *Session) Invalidate() error {
+	s.Flush()
+	return s.Migrate(true)
+}
+
+func (s *Session) Migrate(destroy ...bool) error {
+	shouldDestroy := false
+	if len(destroy) > 0 {
+		shouldDestroy = destroy[0]
+	}
+
+	if shouldDestroy {
+		err := s.driver.Destroy(s.GetID())
+		if err != nil {
+			return err
+		}
+	}
+
+	s.SetID(s.generateSessionID())
+
+	return nil
+}
+
+func (s *Session) Missing(key string) bool {
+	return !s.Exists(key)
+}
+
+func (s *Session) Only(keys []string) map[string]any {
+	return supportmaps.Only(s.attributes, keys...)
+}
+
+func (s *Session) PreviousUrl() string {
+	return s.Get("_previous.url").(string)
+}
+
+func (s *Session) Pull(key string, def ...any) any {
+	return supportmaps.Pull(s.attributes, key, def...)
+}
+
+func (s *Session) Push(key string, value any) sessioncontract.Session {
+	arr := s.Get(key, make([]any, 0)).([]any)
+	arr = append(arr, value)
+	return s.Put(key, arr)
+}
+
 func (s *Session) Put(key string, value any) sessioncontract.Session {
 	s.attributes[key] = value
 	return s
 }
 
+func (s *Session) Regenerate(destroy ...bool) error {
+	err := s.Migrate(destroy...)
+	if err != nil {
+		return err
+	}
+
+	s.RegenerateToken()
+	return nil
+}
+
 func (s *Session) RegenerateToken() sessioncontract.Session {
 	return s.Put("_token", str.Random(40))
 }
```

Comment:
In what scenarios are these two functions used? What's the difference between them?

Reply:
`RegenerateToken` will solely regenerate the token, which can be utilized for CSRF verification. `Regenerate`, on the other hand, optionally deletes the current session.

[Reference](https://laravel.com/docs/10.x/session#regenerating-the-session-id)

![image](https://github.com/goravel/framework/assets/84431594/2e1e2d09-c828-47ad-8829-1482de39ab1e)

Reply:
I see, I noticed that the `RegenerateToken` is a private method in Laravel, at least it doesn't appear in the document, so I suggest we do this, too.

Reply:
Good suggestio, we can make 'RegenerateToken` as a private function.